src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +70,210 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
         })
 }
 
-export function initialize({
+// Add create debugging manifest/requirements.txt containing ptvsd
+const makePythonDebugManifest = async (params: {
+    samProjectCodeRoot: string,
+    outputDir: string
+}): Promise<string | undefined> => {
+    let manifestText = ''
+    const manfestPath = path.join(params.samProjectCodeRoot, 'requirements.txt')
+    if (fileExists(manfestPath)) {
+        manifestText = await readFileAsString(manfestPath)
+    }
+    getLogger().debug(`pythonCodeLensProvider.makePythonDebugManifest params: ${JSON.stringify(params, undefined, 2)}`)
+    if (manifestText.indexOf('ptvsd') < 0 ) { // TODO: Make this logic more robust
+        manifestText += '\nptvsd'
+        const debugManifestPath = path.join(params.outputDir, 'debug-requirements.txt')
+        await writeFile(debugManifestPath, manifestText)
+
+        return debugManifestPath
+    }
+    // else we don't need to override the manifest. nothing to return
+}
+
+// tslint:disable:no-trailing-whitespace
+const makeLambdaDebugFile = async (params: {
+    handlerName: string,
+    debugPort: number,
+    outputDir: string
+}): Promise<{outFilePath: string, debugHandlerName: string}> => {
+    if (!params.outputDir) {
+        throw new Error('Must specify outputDir')
+    }
+    const logger = getLogger()
+
+    const [handlerFilePrefix, handlerFunctionName] = params.handlerName.split('.')
+    const debugHandlerFileName = `${handlerFilePrefix}_debug`
+    const debugHandlerFunctionName = 'lambda_handler'
+    try {
+        logger.info('pythonCodeLensProvider.makeLambdaDebugFile params:', JSON.stringify(params, undefined, 2))
+        const template = `
+import ptvsd
+from ${handlerFilePrefix} import ${handlerFunctionName} as _handler
```

Comment:
Suggest to sanitize/escape all string inputs, to prevent possible injections.

Reply:
Hopefully there is an existing library or regex that we can use. If not, consider constructing a regex from the [lexical (regular) grammar](https://docs.python.org/3/reference/lexical_analysis.html#identifiers).

Reply:
Added a `TODO`. We don't want users code injection attacking their own computers.